code/modules/antagonists/infection/beacon.dm

```diff
@@ -0,0 +1,171 @@
+GLOBAL_LIST_EMPTY(infection_beacons)
+GLOBAL_LIST_EMPTY(beacon_spawns)
+
+/obj/effect/landmark/beacon_start
+	name = "beaconstart"
+	icon_state = "beacon_start"
+
+/obj/effect/landmark/beacon_start/Initialize(mapload)
+	. = ..()
+	GLOB.beacon_spawns += src
+
+/obj/effect/landmark/beacon_start/west
+	name = "beaconstartwest"
+	dir = WEST
+
+/obj/effect/landmark/beacon_start/east
+	name = "beaconstarteast"
+	dir = EAST
+
+/obj/effect/landmark/beacon_start/north
+	name = "beaconstartnorth"
+	dir = NORTH
+
+/obj/effect/landmark/beacon_start/south
+	name = "beaconstartsouth"
+	dir = SOUTH
+
+/*
+	Beacons, infectious creatures and structures are completely unable to cross the barriers that these generate
+*/
+
+/obj/structure/beacon_generator
+	name = "beacon generator"
+	icon = 'icons/mob/infection/infection.dmi'
+	icon_state = "generator"
+	light_range = 4
+	desc = "It sustains the barriers."
+	density = TRUE
+	anchored = TRUE
+	layer = FLY_LAYER
+	CanAtmosPass = ATMOS_PASS_PROC
+	max_integrity = 1000
+	resistance_flags = INDESTRUCTIBLE
+	// Stores the walls that this beacon is generating, to be destroyed when the beacon is destroyed
+	var/list/walls = list()
+
+/obj/structure/beacon_generator/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSobj,src)
+	GLOB.infection_beacons += src
+	update_icon()
+
+/obj/structure/beacon_generator/Destroy()
+	STOP_PROCESSING(SSobj,src)
+	for(var/obj/structure/beacon_wall/D in walls)
+		qdel(D)
+	GLOB.infection_beacons -= src
+	var/mob/camera/commander/C = GLOB.infection_commander
+	C.upgrade_points++
+	C.max_infection_points += 50
+	C.base_point_rate++
+	C.add_points(C.max_infection_points)
+	to_chat(C, "<span class='notice'>You feel pure energy surge through you...</span>")
+	for(var/mob/living/simple_animal/hostile/infection/infectionspore/sentient/spore in C.infection_mobs)
+		spore.add_points(200)
+		to_chat(spore, "<span class='notice'>You feel pure energy surge through you...</span>")
+	if(GLOB.infection_beacons.len > 0)
+		INVOKE_ASYNC(src, .proc/destroyed_announcement)
+	return ..()
+
+/*
+	Delayed announcement that occurs when the becaon has been destroyed
+*/
+/obj/structure/beacon_generator/proc/destroyed_announcement(beacons_left = GLOB.infection_beacons.len)
+	sleep(80)
+	priority_announce("A beacon has been consumed by the infection, only [beacons_left] remain.","CentCom Biohazard Division", 'sound/misc/notice1.ogg')
```

Comment:
Noticed this in the video, should probably read "only 1 remain***s***" when only one beacon remains

Reply:
ah crap i forgot about that thanks mate